src/pybind/mgr/cephadm/tests/test_scheduling.py

```diff
@@ -1753,3 +1753,515 @@ def test_placement_regex_host_pattern(service_type, placement, hosts, expected_a
         daemons=[],
     ).place()
     assert sorted([h.hostname for h in to_add]) == expected_add
+
+
+# Test count-per-label from point where no daemons have been deployed yet
+class CountPerLabelPlacementTestClean(NamedTuple):
+    candidate_hosts: List[HostSpec]
+    labels: List[str]
+    count_per_label: int
+    success: bool
+
+
+@pytest.mark.parametrize("candidate_hosts, labels, count_per_label, success",
+                         [
```

Comment:
I know this is nitpicky, but if you use the "black-style" this won't be so heavily indented and much easier to read (IMO).